### PR TITLE
Add audience API endpoints

### PR DESCRIPTION
**Pull request by toduq**
Resolve https://github.com/line/line-bot-sdk-ruby/issues/172

This pull request adds methods corresponding to the newly added audience API.

The audience API is described here.
https://developers.line.biz/en/docs/messaging-api/using-audience/

This pull request includes 9 methods.

TODO: Check if this code works well

### CODE REVIEW

File: lib/line/bot/client.rb [hunk 1]
```diff
@@ -590,6 +590,127 @@ def get_friend_demographics
...
+        endpoint_path = "/bot/audienceGroup/#{audience_group_id}"
+        get(endpoint, endpoint_path, credentials)
+      end
+
+      # Get data for more than one audience group
+      #
+      # Parameters are described here.
+      # https://developers.line.biz/en/reference/messaging-api/#get-audience-groups
+      #
+      # @param params [Hash] key name `size` is required
```
Comment (kimoto) [inline lib/line/bot/client.rb:681]: optional?
image

File: lib/line/bot/client.rb [hunk 2]
```diff
@@ -590,6 +590,127 @@ def get_friend_demographics
...
+      #
+      # @return [Net::HTTPResponse]
+      def get_audiences(params)
+        channel_token_required
+
+        endpoint_path = "/bot/audienceGroup/list?" + params.map{|k, v| "#{k}=#{v}"}.join('&')
```
Comment (kimoto) [inline lib/line/bot/client.rb:687]: Could you use URI.encode_www_form
```
irb(main):001:0> URI.encode_www_form(a: 'b', c: 'd')
=> "a=b&c=d"
```

File: lib/line/bot/client.rb [hunk 3]
```diff
@@ -590,6 +590,127 @@ def get_friend_demographics
...
+        get(endpoint, endpoint_path, credentials)
+      end
+
+      # Get the authority level of the audience
+      #
```
Comment (kimoto) [inline lib/line/bot/client.rb:692]: Please add api doc url
https://developers.line.biz/en/reference/messaging-api/#get-authority-level

File: lib/line/bot/client.rb [hunk 4]
```diff
@@ -590,6 +590,127 @@ def get_friend_demographics
...
+        channel_token_required
+
+        endpoint_path = "/bot/audienceGroup/authorityLevel"
+        get(endpoint, endpoint_path, credentials)
+      end
+
+      # Change the authority level of the audience
+      #
+      # @param authority_level [String] value must be `PUBLIC` or `PRIVATE`
+      #
```
Comment (kimoto) [inline lib/line/bot/client.rb:704]: https://developers.line.biz/en/reference/messaging-api/#change-authority-level

File: lib/line/bot/client.rb [hunk 5]
```diff
@@ -590,6 +590,127 @@ def get_friend_demographics
...
+      # https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group
+      #
+      # @param opts [Hash] options
+      #
+      # @return [Net::HTTPResponse] This response includes an audience_group_id.
+      def create_user_id_audience(params)
+        channel_token_required
+
+        endpoint_path = '/bot/audienceGroup/upload'
+        post(endpoint, endpoint_path, params.to_json, credentials)
```
Comment (kimoto) [inline lib/line/bot/client.rb:605]: Is it a PUT correctly?
https://developers.line.biz/en/reference/messaging-api/#http-request-20

File: lib/line/bot/client.rb [hunk 6]
```diff
@@ -590,6 +590,127 @@ def get_friend_demographics
...
+      # Rename an existing audience group
+      #
+      # @param audience_group_id [Integer]
+      # @param description [String]
+      #
+      # @return [Net::HTTPResponse]
+      def rename_audience(audience_group_id, description)
+        channel_token_required
+
+        endpoint_path = "/bot/audienceGroup/#{audience_group_id}"
```
Comment (kimoto) [inline lib/line/bot/client.rb:647]: `{audienceGroupId}/updateDescription` ?

**Comment by toduq** [comment]
I've manually checked that this pull request works well.
TODO: test code

File: lib/line/bot/client.rb [hunk 7]
```diff
@@ -641,6 +641,154 @@ def get_bot_info
         get(endpoint, endpoint_path, credentials)
       end
 
+      # Create an audience group by uploading user_ids
+      #
+      # Parameters are described here.
+      # https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group
+      #
+      # @param opts [Hash] options
```
Comment (kimoto) [inline lib/line/bot/client.rb:649]: params?

**Review by kimoto — APPROVED** [review]

**Pull request merged**